### PR TITLE
Feature/861/131 fix sample 043

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

- Fixed sample 04.3 that was broken (env variables should now be in UPPER_CASE; issue with ports/API Key configuration)
- Used `jdk-logger-monitor` in the sample so that monitor logs are exported to java.util.logging, and from there captured by the App Insights agent if used
- Fixed instructions for using Application Insights with the sample

## Why it does that

- Sample was broken
- Need sample for App Insights

## Further notes

See sample transaction at https://ms.portal.azure.com/#blade/AppInsightsExtension/DetailsV2Blade/ComponentId/%7B%22SubscriptionId%22%3A%229d236f09-93d9-4f41-88a6-20201a6a1abc%22%2C%22ResourceGroup%22%3A%22algattik-appinsights%22%2C%22Name%22%3A%22algattik-appinsights%22%2C%22LinkedApplicationType%22%3A0%2C%22ResourceId%22%3A%22%252Fsubscriptions%252F9d236f09-93d9-4f41-88a6-20201a6a1abc%252Fresourcegroups%252Falgattik-appinsights%252Fproviders%252Fmicrosoft.insights%252Fcomponents%252Falgattik-appinsights%22%2C%22ResourceType%22%3A%22microsoft.insights%252Fcomponents%22%2C%22IsAzureFirst%22%3Afalse%7D/DataModel/%7B%22eventId%22%3A%223e6c4536-a53e-11ec-a999-00155da5c821%22%2C%22timestamp%22%3A%222022-03-16T15%3A31%3A51.140Z%22%2C%22cacheId%22%3A%223123244d-83e3-4b02-9fbf-62eb9034eefd%22%2C%22eventTable%22%3A%22traces%22%2C%22timeContext%22%3A%7B%22durationMs%22%3A14400000%2C%22endTime%22%3A%222022-03-16T17%3A30%3A32.988Z%22%7D%7D

## Linked Issue(s)

Relates to #131

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: core/contract/README.md [hunk 1]
```diff
@@ -1,2 +1,11 @@
-# Contract Extension 
+# Contract
```
Comment (cpeeyush) [inline core/contract/README.md:1]: Looks like this change is not related to this PR. Do we want to do it here ?

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/AbstractContractNegotiationManager.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (cpeeyush) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/AbstractContractNegotiationManager.java:2]: same here, is this change related to this PR ? Is it due to some merge issue ?

Looks like this PR has many changes like this.

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java [hunk 3]
```diff
@@ -60,6 +62,12 @@
     private ConsumerContractNegotiationManagerImpl consumerNegotiationManager;
     private ProviderContractNegotiationManagerImpl providerNegotiationManager;
 
+    @EdcSetting
```
Comment (cpeeyush) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java:65]: is this change related to this PR ?

**Review by cpeeyush — CHANGES_REQUESTED** [review]

**Comment by algattik** [comment]
Replaced with #167
